Armadeus APF modules require some special U-Boot config variables to be set.

Index: buildroot/boot/u-boot/u-boot.mk
===================================================================
--- buildroot.orig/boot/u-boot/u-boot.mk	2011-08-08 12:13:56.000000000 +0200
+++ buildroot/boot/u-boot/u-boot.mk	2011-08-08 12:14:20.000000000 +0200
@@ -139,6 +139,17 @@
 endif
 	@echo "#endif /* __BR2_ADDED_CONFIG_H */" >> $(U_BOOT_INC_CONF_FILE)
 endif # BR2_TARGET_UBOOT_NETWORK
+ifneq ($(strip $(BOARD_NAME)),"")
+	$(call insert_define, CONFIG_BOARD_NAME, $(BOARD_NAME))
+endif
+ifneq ($(BR2_TARGET_ARMADEUS_SDRAM_SIZE),)
+	$(SED) "s,^#define CFG_SDRAM_MBYTE_SYZE.*,#define CFG_SDRAM_MBYTE_SYZE $(BR2_TARGET_ARMADEUS_SDRAM_SIZE),g;"\
+		$(U_BOOT_INC_CONF_FILE)
+endif
+ifneq ($(BR2_TARGET_ARMADEUS_NR_SDRAM),)
+	$(SED) "s,^#define CONFIG_NR_DRAM_BANKS.*,#define CONFIG_NR_DRAM_BANKS $(BR2_TARGET_ARMADEUS_NR_SDRAM),g;"\
+		$(U_BOOT_INC_CONF_FILE)
+endif
 	touch $@
 
 # Build U-Boot itself
